sql/tables.go

```diff
@@ -135,6 +135,20 @@ type IndexAlterableTable interface {
 	RenameIndex(ctx *Context, fromIndexName string, toIndexName string) error
 }
 
+// IndexBuildingTable is an optional extension to IndexAlterableTable that supports the engine's assistance in building
+// a newly created index, or rebuilding an existing one. This interface is non-optional for tables that wish to create
+// indexes on virtual columns, as the engine must provide a value for these columns.
+type IndexBuildingTable interface {
```

Comment:
this kind of seems like the main change, code for it doesn't look full functional yet though. my instinct is that an "optional but not optional" interface with naming as broad as "BuildIndex" sounds confusing. Should it be named `BuildVirtualIndex`? `buildProjectionIndex`? I'm assuming it's a performance hit for this to be required (bytes->sql.Row-?bytes)?

Reply:
Kind of premature, defined the interface before I realized I didn't strictly need it to get virtual columns working. We do a full table rewrite on index creation, and it would be better to just build the index being added.